### ISSUE TITLE
Unable to get @Value or @Autowired for @Providers

### ISSUE THREAD

**Issue by trajano** [paypal/resteasy-spring-boot#81]
I have a class that looks like this...

```
@Configuration
@Component
@Provider
public class JsonExceptionMapper implements
    ExceptionMapper<Throwable> {

    @Autowired
    Environment environment;

    @Value("${microservice.show_stack_trace}")
    private boolean showStackTrace;
```

However, when I get the value for `showStackTrace` it is always false.  Also anything `@Autowired` returns null

I am registering this class via `MyApplication.getClasses()`

**Comment by fabiocarvalho777** [comment]
Hello,
Do not register this class via `MyApplication.getClasses()`.
Instead of that, just make sure it is a Spring bean, that is enough.
I see it is annotated as `@Component`. In this case, then just make sure its package is scanned by Spring, and then everything should work fine.

**Comment by trajano** [comment]
Actually it does not work.  If I don't have it in MyApplication.getClasses() the provider does not even register and just uses the default 404 handler.  I tried to register it a few different ways

```
@ComponentScan
@SpringBootApplication(scanBasePackageClasses = {JsonExceptionMapper.class, ExampleApplication.class})

        SpringApplication.run(new Class[]{
            ExampleApplication.class,
JsonExceptionMapper.class
        }, args);
```

**Comment by fabiocarvalho777** [comment]
Hello @trajano ,

Just to prove it's been working as designed, I have just added to the [sample app](https://github.com/paypal/resteasy-spring-boot/tree/develop/sample-app) an ExceptionMapper which has an injection point. I have also added an automated integration test to exercise that provider (the exception mapper) and everything worked normally.

[You can see what I did in this commit](https://github.com/paypal/resteasy-spring-boot/commit/e96c9e4c03acf035600ca715a8950ea85a862df7).

I recommend you study the sample application and double check your application to find what is causing your problem.

Have a nice day.

**Comment by trajano** [comment]
Thanks will do. One difference we had just at cursory glance was you don't use @Configuration and @Value but you did have Autowired and component so I will see how things fare with that. 

One other thing that is likely different is my exception mapper is on a different jar

**Comment by fabiocarvalho777** [comment]
> One other thing that is likely different is my exception mapper is on a different jar

I have many applications where providers come from different jars and it also works fine. So, no, that is not the case neither.

I believe the reason is still the fact that your class is not being registered as a Spring bean properly.

Try this:

```java
@ComponentScan({"com.package.to.your.exception.mapper.class"})
@SpringBootApplication
```

Notice that I removed any parameter from `SpringBootApplication` annotation and added to `ComponentScan`.

Also, you do NOT have to add explicitly to `ComponentScan` any class that is under the package where your entry point Spring Boot class is. Those classes are scanned automatically. [That is why I don't have a `ComponentScan` annotation here](https://github.com/paypal/resteasy-spring-boot/blob/master/sample-app/src/main/java/com/sample/app/Application.java#L13).

**Comment by trajano** [comment]
Thanks for the tips.  Hopefully it's just that when I try it at work tomorrow.  (Sorry forgot to write that I was on the bus when I was replying)

However, I did recall something that may be causing the problem or at least hiding the problem.  The `@Value` annotation I made it point to a `boolean` I think it may work if I change it to `Boolean`.  Anyway I'll look at the code again tomorrow and see if there's anything amiss in comparison to yours.
